tensorboard/webapp/runs/views/runs_table/runs_table_container.ts

```diff
@@ -440,36 +440,14 @@ export class RunsTableContainer implements OnInit, OnDestroy {
           return items;
         }
 
-        let regex: RegExp | null = null;
-
-        // Do not break all the future updates because of malformed
-        // regexString. User can be still modifying it.
-        try {
-          regex = regexString ? new RegExp(regexString) : null;
-        } catch (e) {}
-
-        if (!regex) {
-          return [];
-        }
-
         const shouldIncludeExperimentName = this.columns.includes(
           RunsTableColumn.EXPERIMENT_NAME
         );
         return items.filter((item) => {
-          // For some reason, TypeScript is faulty and cannot coerce the type to
-          // non-null inspite the `if (!regex)` above.
-          regex = regex!;
-
-          if (!shouldIncludeExperimentName) {
-            return regex.test(item.run.name);
-          }
-
-          const legacyRunName = `${item.experimentName}/${item.run.name}`;
-          return (
-            regex.test(item.run.name) ||
-            regex.test(item.experimentAlias) ||
-            regex.test(item.experimentName) ||
-            regex.test(legacyRunName)
+          return matchRunToRegex(
+            {...item.run, ...item},
```

Comment:
It may be logically messy but type wise, I think it is okay for users of the API to provide superset as long as it is strongly type enforced by the compiler.

Still, I've made some changes to explicitly require only very tight subset of the Run that I require in the method.